File: backend/src/main/java/org/springframework/boot/backend/service/AlulaService.java
package org.springframework.boot.backend.service;

import org.springframework.boot.backend.entity.Alula;

import java.util.List;
import java.util.Optional;

public interface AlulaService {
    List<Alula> getAllAlulas();
    Optional<Alula> getAlulaById(Long id);
}

File: frontend/src/app/services/attributes/alula.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Alula } from '../../interfaces/attributes/alula';

@Injectable({
  providedIn: 'root'
})
export class AlulaService {
  private apiUrl = 'http://localhost:8080/alula'; // Update with your backend API URL

  constructor(private http: HttpClient) {}

  getAllAlulas(): Observable<Alula[]> {
    return this.http.get<Alula[]>(this.apiUrl);
  }

  getAlulaById(id: number): Observable<Alula> {
    return this.http.get<Alula>(`${this.apiUrl}/${id}`);
  }
}